client/transport/stdio_test.go

```diff
@@ -297,14 +312,75 @@ func TestStdio(t *testing.T) {
 		if responseError.Method != "debug/echo_error_string" {
 			t.Errorf("Expected method 'debug/echo_error_string', got '%s'", responseError.Method)
 		}
-		if responseError.ID != 100 {
-			t.Errorf("Expected ID 100, got %d", responseError.ID)
+		idValue, ok := responseError.ID.Value().(int64)
+		if !ok {
+			t.Errorf("Expected ID to be int64, got %T", responseError.ID.Value())
+		} else if idValue != 100 {
+			t.Errorf("Expected ID 100, got %d", idValue)
 		}
 		if responseError.JSONRPC != "2.0" {
 			t.Errorf("Expected JSONRPC '2.0', got '%s'", responseError.JSONRPC)
 		}
 	})
 
+	t.Run("SendRequestWithStringID", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+		defer cancel()
+
+		params := map[string]any{
+			"string": "string id test",
+			"array":  []any{4, 5, 6},
+		}
+
+		// Use a string ID instead of an integer
+		request := JSONRPCRequest{
+			JSONRPC: "2.0",
+			ID:      mcp.NewRequestId("request-123"),
+			Method:  "debug/echo",
+			Params:  params,
+		}
```

Comment:
Make sure string Ids work. Only this test uses a string ID now but I guess that's sufficient.